python/pyarrow/tests/test_scalars.py

```diff
@@ -219,6 +220,9 @@ def test_bool():
     assert true.as_py() is True
     assert false.as_py() is False
 
+    assert bool(true) is True
+    assert bool(false) is False
```

Comment:
Perhaps also test the case where the boolean isn't valid?
Something like:
```suggestion
    assert bool(false) is False
    
    null = pa.scalar(None, type=pa.bool_())
    assert isinstance(null, pa.BooleanScalar)
    # other tests to be added
```

Reply:
Great idea, also added